### PR TITLE
[WIP] Add headers and navigation

### LINKED ISSUE #289: Create coded version of Header + Nav
**Description**
We currently only have a placeholder image version of Headers + Nav. This issue is to complete the work to create an actual coded, and responsive, version of the visual design and any UX wireframes (see layout below).

**To Do**
- [ ] Create html and styles (responsive) based off of current structure
- [ ] Run HTML Codesniffer and review for accessibility quality and make any necessary changes
- [ ] Submit pull request when code is ready for review
- [ ] Once reviewed and merged into staging, run by @angel/@mollieru and @carodew/@colinpmacarthur for final approval.
- [ ] Iterate/make changes and re-submit PR with any fixes prior to pushing to production

**Visual Design**
headers_navigation_ui_v2

### LINKED ISSUE #705: [Header+ top navigation] design draft
Our second iteration of a Header layout with top navigation design for desktop and mobile, not explicitly included in v.8 release:

_Tip: click on image for full size_
headers_navigation_ui_v3

### PR DESCRIPTION
**Pull request by maya**
WIP :construction: :warning: 

This adds the header and side navigation to the site.

This resolves #705.

### CODE REVIEW

**Comment by jeremiak** [comment]
Some feedback from trying to implement this on [jeremiak/sbst-template](https://github.com/jeremiak/sbst-template):

The grid system inside the nav bar isn't the most intuitive to me. For example, I have the following code for putting the site title in the upper left hand corner:

```
<a href="/" accesskey="1" aria-label="Home">
    <h1 class="usa-header-title">{{ site.title }}</h1>
</a>
```

But the problem is that the text wraps after just the first word (screenshot [here](http://up.jeremiak.com/cGg8)).

It still happens even when the number of menu items is fewer, though still a reasonable 4 (screenshot [here](http://up.jeremiak.com/cGh0)).

I'm used to working with frameworks like Bootstrap that are pretty forgiving in how the `.navbar` works and doesn't rely on an internal grid system, instead relying more on having a logo/title and then a flexible list of menu items.

**Comment by maya** [comment]
@jeremiak yeah we might just change that so there's no grid in the header, especially not for the logo. I was just following the design comps initially which indicated a grid there.

**Comment by jeremiak** [comment]
Ya, from my perspective I'd very much like the grid to not have to be as present as it is. It seems to mostly get in my way and not be easily to debug. So for the main content page, I get using `.usa-grid` to create a new grid and then using the column classes and `.end-row` to get the effect I want.

However, for the navbar I really just want to make a title and then make a UL with my menu items.

**Comment by maya** [comment]
@jeremiak my last commits remove the grid classes from the logo + nav list. Let me know if that works out better for you.

File: assets/_scss/components/_header.scss [hunk 1]
```diff
@@ -0,0 +1,374 @@
+header[role="banner"] {
```
Comment (msecret) [inline assets/_scss/components/_header.scss]: I'm not sure this is the best selector to encompass the header. One thing: do you know if this selector will work in other browsers? Besides that, as a developer, I wouldn't ever suspect a role attribute to effect styling. Another thing, this a strong selector (one lvl1 plus one lvl2). Is there a reason we shouldn't just make this a class?

Comment (maya) [inline assets/_scss/components/_header.scss]: @msecret I'm fine with making this a class.

File: assets/_scss/components/_header.scss [hunk 2]
```diff
@@ -0,0 +1,374 @@
...
+
+  .usa-grid {
+    padding: 0;
+
+    @include media($medium-screen) {
+      padding: 0 2.5rem;
+    }
+  }
+
+  .usa-logo-block {
```
Comment (msecret) [inline assets/_scss/components/_header.scss]: Why not just `.usa-logo`?

File: _components/headers.md [hunk 3]
```diff
@@ -5,8 +5,453 @@ title: Headers & Navigation
...
+        <a class="usa-menu-btn" id="menu-btn" href="#">Menu</a>
+        <div class="usa-logo-block" id="logo">
+          <a href="javascript:void(0)" accesskey="1" aria-label="Home">
+            <h1 class="usa-logo-title">Department of Web Standards</h1>
+          </a>
+        </div>
+        <a class="usa-search-trigger" href="#">
+        </a>
+        <nav class="usa-nav-primary" role="navigation">
+          <ul class="usa-nav-list usa-unstyled-list">
```
Comment (msecret) [inline _components/headers.md]: I think we should use a mixin here for `usa-unstyled-list`

File: assets/_scss/components/_header.scss (hunk 1 above)
Comment (maya) [inline assets/_scss/components/_header.scss]: @msecret done in d496811.

File: _components/headers.md (hunk 3 above)
Comment (maya) [inline _components/headers.md]: Done in efc8d91.

File: assets/_scss/components/_header.scss (hunk 2 above)
Comment (maya) [inline assets/_scss/components/_header.scss]: @msecret done in efc8d91.

File: _components/headers.md (hunk 3 above)
Comment (maya) [inline _components/headers.md]: @msecret do you think we should go through the library and do the same thing for all the places we've used the `usa-unstyled-list` class?

I think this is much cleaner markup.

**Comment by maya** [comment]
This has gotten so far out of date. Closing in place of #832.

**Pull request closed without merging**